### PR TITLE
feat: build process

### LINKED ISSUE #1: Building process
Add building process to generate and publish package.

### PR DESCRIPTION
**Pull request by miguelramos**
Added script to generate esm module and iife. Closes #1

### CODE REVIEW

File: src/schema-types/string.schema-type.ts [hunk 1]
```diff
@@ -1,3 +1,10 @@
+/**
+ * Copyright Websublime All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://websublime.dev/license
+ */
+
```
Comment (miguelramos) [inline src/schema-types/string.schema-type.ts:7]: @Hulago should we check for alternative on the regular expression?

**Comment by miguelramos** [comment]
@Hulago is ready for review

File: src/schema-types/string.schema-type.ts (hunk 1 above)
Comment (Hulago) [reply src/schema-types/string.schema-type.ts:7]: `
function isValidUrl(string) {  
  try {
    url = new URL(string);
    return true;
  } catch (_) {
    return false;  
  }
}
`

And this alternative code. It also works in node.

https://caniuse.com/url

Comment (Hulago) [reply src/schema-types/string.schema-type.ts:7]: We may need to add additional validation on the protocol.

Comment (miguelramos) [reply src/schema-types/string.schema-type.ts:7]: Hmm, protocol is really needed. Wrong things can be passed to this.

image

Comment (miguelramos) [reply src/schema-types/string.schema-type.ts:7]: This alternative could be the answer. What you think?

```js
const isValidUrl = (url, matchProtocol = false) => {
  const regex = `(http(s)?:\/\/.)${matchProtocol ? '' : '?'}(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)`;
  const expression = new RegExp(regex, 'g');

  return expression.test(url);
};
```

image

Comment (Hulago) [reply src/schema-types/string.schema-type.ts:7]: The thing is protocol can be anything. mqtt, ssh, ftp, mailto, ws, wss, xpto.
Using the URL object we will be compliant with what is a valid URL for javascript.

The if we want, we can create another specific rules like isValidHTTPUrl, with a regexp on the protocol property of the URL object.

Comment (miguelramos) [reply src/schema-types/string.schema-type.ts:7]: Ok, so we have validations for particular things like protocol way. I agree will be more friendly.

Comment (miguelramos) [reply src/schema-types/string.schema-type.ts:7]: Will be applied on PR #5 . Closing to continue.

**Pull request merged**